src/compact.jl

```diff
@@ -112,14 +121,15 @@ macro compact(fex, _kwexs...)
   # make strings
   layer = "@compact"
   setup = NamedTuple(map(ex -> Symbol(string(ex.args[1])) => string(ex.args[2]), kwexs))
-  input = join(fex.args[1].args, ", ")
+  input = try join(fex.args[1].args, ", ") catch e 
+    @warn "function stringifying does not yet handle all cases, using empty string"
```

Comment:
Would be useful to print the error message here as well. For example the following lambda seems to break something:

```julia
julia> model = @compact(x -> a * x; a=rand())
┌ Warning: function stringifying does not yet handle all cases, using empty string
└ @ Fluxperimental /private/var/folders/1h/xyppkvx52cl6w3_h8bw_gdqh0000gr/T/tmp.lEuAu2Ch/Fluxperimental.jl/src/compact.jl:125
@compact(
  a = rand(),
)
    a * x
end
```